packages/core/content-releases/server/src/controllers/release-action.ts

```diff
@@ -17,6 +20,62 @@ const releaseActionController = {
       data: releaseAction,
     };
   },
+  async findMany(ctx: Koa.Context) {
+    const releaseId: GetReleaseActions.Request['params']['releaseId'] = ctx.params.releaseId;
+
+    const allowedContentTypes = getAllowedContentTypes({ strapi, userAbility: ctx.state.userAbility });
+    
+    const morphSanitizedPopulate = await allowedContentTypes.reduce(async (promiseAccumulator, contentTypeUid) => {
+      const accumulator = await promiseAccumulator;
+
+      const permissionsManager = await strapi.admin.services.permission.createPermissionsManager({
+        ability: ctx.state.userAbility,
+        model: contentTypeUid
+      });
+
+      // We create this to filter populated entries based on custom conditions on permissions
+      accumulator[contentTypeUid] = {
+        filters: permissionsManager.getQuery(CONTENT_MANAGER_READ_ACTION)
+      };
+
+      return accumulator;
+    }, Promise.resolve({} as Record<UID.ContentType, { filters: object }>));
+
+
+    const releaseService = getService('release', { strapi });
+    const { results, pagination } = await releaseService.findActions(releaseId, allowedContentTypes, { 
+      populate: {
+        entry: {
+          on: morphSanitizedPopulate
+        }
+      }
+    });
+
+    // Because this is a morphTo relation, we need to sanitize each entry separately based on its contentType
+    const sanitizedResult = await Promise.all(results.map(async (action: ReleaseAction) => {
```

Comment:
`sanitizedResults`? 🤷‍♂️

Reply:
same as for the reduce async! there is a map async in strapi/utils